pycbc/events/simd_threshold_ccode.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ *  Copyright (C) 2014 Josh Willis
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License as published by the
+ *  Free Software Foundation; either version 3 of the License, or (at your
+ *  option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ *  Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/*
+ * This module defines several C functions that compute
+ * a combined thresholding and time clustering of a complex array using a
+ * multithreaded, SIMD vectoried code.
+ *
+ * There are four C functions defined:
+ *
+ * parallel_threshold: A multithreaded function that will identify points in a
+ *                     timeseries above a user-input threshold.
+ *
+ * max_simple: A single-threaded function that does NOT use explicit 
+ *             SIMD vectorization,
+ *             to compute the maximum of a complex time series over a given
+ *             window.
+ *
+ * windowed_max: A single threaded (but not vectorized) function that finds the
+ *               locations, norms, and complex values of the maxima in each of
+ *               a set of predefined windows in a given complex array. It does
+ *               this by calling max_simple on each window with the
+ *               appropriate parameters.
+ *
+ * parallel_thresh_cluster: A multithreaded function that finds the maxima in
+ *                          each of a set of contiguous, fixed-length windows
+ *                          by calling windowed_max in parallel. It then sweeps
+ *                          through the results of that (single-threaded) and
+ *                          tests for above threshold, and time-clusters the
+ *                          surviving triggers.
+ *
+ * A user calls only the last function; the other three exist to conveniently
+ * compartmentalize SIMD code from OpenMP code.
+ */
+
+#include <omp.h> // FIXME: Does this need to be optional for clang??
```

Comment:
While the pragmas I believe will be ignore, I think this import yes may post a problem for clang on a default Mac setup. Unless they are now enabling omp by default?

Reply:
Ok, googling says that OMP is now *enabled* in clang 3.8.0 and later. I'm not sure what Mac versions that corresponds to.

Reply:
Ah, nvm, looks like Mac overrides the clang defaults and disables it...

Reply:
I think one can put this in a macro ala https://stackoverflow.com/questions/28166565/detect-gcc-as-opposed-to-msvc-clang-with-macro

So something like
```
#ifndef __clang__
    #include <omp.h>
#endif
```

Reply:
I'll make sure this builds on my mac before merging!

Reply:
@spxiwh You'll want to make sure you are using system clang and not from macports or elsewhere. I think those might actually have openmp. Add it to the list to stand up a MaxOSX test build at some point in travis....

Reply:
Well without these lines this doesn't build on my mac and with them it compiles, so I'm adding them and removing the FIXME. (This becomes additionally confusing as mac symlinks `gcc` to `clang` so even though it looks like you're using gcc, you're not)